### ISSUE TITLE
[Feature Request] Will Mojo natively support computing gradients and backward in the future?

### ISSUE THREAD

**Issue by Jordan-Haidee** [modularml/mojo#660]
### Review Mojo's priorities

- [x] I have read the [roadmap and priorities](https://docs.modular.com/mojo/roadmap.html#overall-priorities) and I believe this request falls within the priorities.

### What is your request?

- In Python, if I want to compute gradient and train a network, I have to install third-party package like jax/pytorch/tenorflow etc. 

- SO I want to ask whether Mojo will natively support gradient and backward from the underlying architecture design.

### What is your motivation for this change?

I noticed that Mojo's slogan is the programming language of the AI, so I would like to know the answer to the above question.

**Comment by lattner** [comment]
I have experience building SCT autodiff into Swift:
https://github.com/apple/swift/blob/main/docs/DifferentiableProgramming.md

we may investigate something like that in the distant future, but don't have plans short term.  Until then, library based approaches should work fine in Mojo
